File: ex11/ex11_03/ex11_03.py
from datetime import datetime

def main():
  print("Enter a schedule date in the format YYYY-MM-DD HH:MM:SS")
  schedule_date = input("Date: ").strip()
  if len(schedule_date) == 10:
    schedule_date += " 00:00:00"
  elif len(schedule_date) == 13:
    schedule_date += ":00:00"
  elif len(schedule_date) == 16:
    schedule_date += ":00"
  schedule_date = datetime.strptime(schedule_date, "%Y-%m-%d %H:%M:%S")
  current_date = datetime.now()
  print(current_date)
  time_left = schedule_date - current_date
  if schedule_date < current_date:
    print("The schedule date is passed!")
    return
  str_time = "Time left: " + str(time_left.days) + " days, " + str(time_left.seconds // 3600) + " hours, "
  str_time += str((time_left.seconds // 60) % 60) + " minutes, " + str(time_left.seconds % 60) + " seconds"
  print(str_time)

if __name__ == "__main__":
  main()

File: ex11/ex11_03/ex11_03.cpp
#include <iostream>
#include "include/mytime.h"
using namespace std;
  
int main() {
  char schedule_date_str[50];
  struct tm tm_plan = {};
  struct tm tm_left = {};
  cout << "Enter a schedule date in the format YYYY-MM-DD HH:MM" << endl;
  cout << "Date: ";
  fgets(schedule_date_str, sizeof(schedule_date_str), stdin);
  schedule_date_str[strlen(schedule_date_str)-1] = '\0';
  if (strlen(schedule_date_str) == 10 ) {
    strcat(schedule_date_str, " 00:00:00");
  }
  else if (strlen(schedule_date_str) == 13) {
    strcat(schedule_date_str, ":00:00");  
  }
  else if (strlen(schedule_date_str) == 16) {  
    strcat(schedule_date_str, ":00");
  }
  strptime(schedule_date_str, "%Y-%m-%d %H:%M:%S", &tm_plan);
  tm_plan.tm_isdst = -1;
  time_t schedule_date = mktime(&tm_plan);
  time_t current_date = time(nullptr);
  double time_left = difftime(schedule_date, current_date);
  if (time_left < 0) {
    cout << "The schedule date was passed!" << endl;
    return 0;
  }
  tm_left = get_time_left(time_left);
  cout << "Time left: " 
    << tm_left.tm_mday << " days, "
    << tm_left.tm_hour << " hours, "
    << tm_left.tm_min << " minutes, "
    << tm_left.tm_sec << " seconds" << endl;
  return 0;
}